app/controllers/support_interface/provider_users_controller.rb

```diff
@@ -10,11 +10,23 @@ def new
 
     def create
       @form = ProviderUserForm.new(provider_user_params)
-
-      if @form.save
-        flash[:success] = 'Provider user created'
-        redirect_to support_interface_provider_users_path
-      else
+      provider_user = @form.build
+
+      if @form.valid?
+        service = InviteProviderUser.new(provider_user: provider_user)
+        begin
+          if service.save_and_invite!
+            flash[:success] = 'Provider user created'
+            redirect_to support_interface_provider_users_path
+          else # show errors from model
```

Comment:
If the form's declared the model valid, would this ever happen?

Reply:
The only time this can happen is if the validations from the model and the form diverge. Currently, this could happen with the model's unique validation on `dfe_sign_in_uid`, the likelihood of which is non-existent. But I left  this `else` there as a catch-all for non-exception errors.